### PR TITLE
feat: add GKE exceptions

### PR DESCRIPTION
**Pull request by pbr0ck3r**
kor version: vdev

  _  _____  ____
 | |/ / _ \|  _ \
 | ' / | | | |_) |
 | . \ |_| |  _ <
 |_|\_\___/|_| \_\

Unused Resources in Namespace: gmp-system
+---+---------------+---------------------------+
| # | RESOURCE TYPE |       RESOURCE NAME       |
+---+---------------+---------------------------+
| 1 | ConfigMap     | config-images             |
| 2 | ConfigMap     | webhook-ca                |
| 3 | ReplicaSet    | rule-evaluator-6fd7ddb94c |
+---+---------------+---------------------------+

Unused Resources in Namespace: kube-system
+----+----------------+------------------------------------------------------+
| #  | RESOURCE TYPE  |                    RESOURCE NAME                     |
+----+----------------+------------------------------------------------------+
|  1 | ConfigMap      | cluster-autoscaler-status                            |
|  2 | ConfigMap      | cluster-kubestore                                    |
|  3 | ConfigMap      | clustermetrics                                       |
|  4 | ConfigMap      | extension-apiserver-authentication                   |
|  5 | ConfigMap      | gke-common-webhook-heartbeat                         |
|  6 | ConfigMap      | ingress-uid                                          |
|  7 | ConfigMap      | konnectivity-agent-autoscaler-config                 |
|  8 | ConfigMap      | kube-apiserver-legacy-service-account-token-tracking |
|  9 | ConfigMap      | kube-dns-autoscaler                                  |
| 10 | ConfigMap      | kubedns-config-images                                |
| 11 | Service        | vpa-recommender                                      |
| 12 | ServiceAccount | metadata-proxy                                       |
| 13 | Role           | cloud-provider                                       |
| 14 | Role           | system:controller:glbc                               |
| 15 | ReplicaSet     | metrics-server-v0.6.3-7cb4458849                     |
| 16 | DaemonSet      | fluentbit-gke-256pd                                  |
| 17 | DaemonSet      | fluentbit-gke-max                                    |
| 18 | DaemonSet      | gke-metrics-agent-scaling-10                         |
| 19 | DaemonSet      | gke-metrics-agent-scaling-100                        |
| 20 | DaemonSet      | gke-metrics-agent-scaling-20                         |
| 21 | DaemonSet      | gke-metrics-agent-scaling-200                        |
| 22 | DaemonSet      | gke-metrics-agent-scaling-50                         |
| 23 | DaemonSet      | gke-metrics-agent-scaling-500                        |
| 24 | DaemonSet      | gke-metrics-agent-windows                            |
| 25 | DaemonSet      | kube-proxy                                           |
| 26 | DaemonSet      | metadata-proxy-v0.1                                  |
| 27 | DaemonSet      | nccl-fastsocket-installer                            |
| 28 | DaemonSet      | nvidia-gpu-device-plugin-large-cos                   |
| 29 | DaemonSet      | nvidia-gpu-device-plugin-large-ubuntu                |
| 30 | DaemonSet      | nvidia-gpu-device-plugin-medium-cos                  |
| 31 | DaemonSet      | nvidia-gpu-device-plugin-medium-ubuntu               |
| 32 | DaemonSet      | nvidia-gpu-device-plugin-small-cos                   |
| 33 | DaemonSet      | nvidia-gpu-device-plugin-small-ubuntu                |
| 34 | DaemonSet      | pdcsi-node-windows                                   |
| 35 | DaemonSet      | runsc-metric-server                                  |
| 36 | DaemonSet      | tpu-device-plugin                                    |
+----+----------------+------------------------------------------------------+

Unused Crds: 
+----+---------------+-------------------------------------------------+
| #  | RESOURCE TYPE |                  RESOURCE NAME                  |
+----+---------------+-------------------------------------------------+
|  1 | Crd           | allowlistedv2workloads.auto.gke.io              |
|  2 | Crd           | allowlistedworkloads.auto.gke.io                |
|  3 | Crd           | backendconfigs.cloud.google.com                 |
|  4 | Crd           | capacityrequests.internal.autoscaling.gke.io    |
|  5 | Crd           | clusterpodmonitorings.monitoring.googleapis.com |
|  6 | Crd           | clusterrules.monitoring.googleapis.com          |
|  7 | Crd           | frontendconfigs.networking.gke.io               |
|  8 | Crd           | gkenetworkparamsets.networking.gke.io           |
|  9 | Crd           | globalrules.monitoring.googleapis.com           |
| 10 | Crd           | managedcertificates.networking.gke.io           |
| 11 | Crd           | memberships.hub.gke.io                          |
| 12 | Crd           | networks.networking.gke.io                      |
| 13 | Crd           | podmonitorings.monitoring.googleapis.com        |
| 14 | Crd           | provisioningrequests.autoscaling.x-k8s.io       |
| 15 | Crd           | rules.monitoring.googleapis.com                 |
| 16 | Crd           | serviceattachments.networking.gke.io            |
| 17 | Crd           | servicenetworkendpointgroups.networking.gke.io  |
| 18 | Crd           | updateinfos.nodemanagement.gke.io               |
| 19 | Crd           | volumesnapshotclasses.snapshot.storage.k8s.io   |
| 20 | Crd           | volumesnapshotcontents.snapshot.storage.k8s.io  |
| 21 | Crd           | volumesnapshots.snapshot.storage.k8s.io         |
+----+---------------+-------------------------------------------------+

Unused ClusterRoles: 
+----+---------------+----------------------------------------------------------------------+
| #  | RESOURCE TYPE |                            RESOURCE NAME                             |
+----+---------------+----------------------------------------------------------------------+
|  1 | ClusterRole   | admin                                                                |
|  2 | ClusterRole   | cloud-provider                                                       |
|  3 | ClusterRole   | edit                                                                 |
|  4 | ClusterRole   | system:aggregate-to-admin                                            |
|  5 | ClusterRole   | system:aggregate-to-edit                                             |
|  6 | ClusterRole   | system:certificates.k8s.io:certificatesigningrequests:nodeclient     |
|  7 | ClusterRole   | system:certificates.k8s.io:certificatesigningrequests:selfnodeclient |
|  8 | ClusterRole   | system:certificates.k8s.io:kube-apiserver-client-approver            |
|  9 | ClusterRole   | system:certificates.k8s.io:kube-apiserver-client-kubelet-approver    |
| 10 | ClusterRole   | system:certificates.k8s.io:kubelet-serving-approver                  |
| 11 | ClusterRole   | system:certificates.k8s.io:legacy-unknown-approver                   |
| 12 | ClusterRole   | system:controller:cloud-node-controller                              |
| 13 | ClusterRole   | system:controller:glbc                                               |
| 14 | ClusterRole   | system:heapster                                                      |
| 15 | ClusterRole   | system:kube-aggregator                                               |
| 16 | ClusterRole   | system:kubelet-api-admin                                             |
| 17 | ClusterRole   | system:persistent-volume-provisioner                                 |
+----+---------------+----------------------------------------------------------------------+

Unused StorageClasss: 
+---+---------------+---------------+
| # | RESOURCE TYPE | RESOURCE NAME |
+---+---------------+---------------+
| 1 | StorageClass  | premium-rwo   |
| 2 | StorageClass  | standard      |
| 3 | StorageClass  | standard-rwo  |
+---+---------------+---------------+

### CODE REVIEW

File: pkg/kor/replicaset.go [hunk 1]
```diff
@@ -13,6 +13,19 @@ import (
 	"github.com/yonahd/kor/pkg/filters"
 )
 
+// These replica sets generated for default GKE
+// TODO: Need to add it the ability for isResourceException to handle regex/wildcard matching
```
Comment (yonahd) [inline pkg/kor/replicaset.go:17]: GKE creates empty replicasets?

Comment (pbr0ck3r) [reply pkg/kor/replicaset.go:17]: Yeah, I just re-created a new cluster again and these are not there. Going to remove them.

Comment (pbr0ck3r) [reply pkg/kor/replicaset.go:17]: Depending on if you build a `standard` cluster or use `autopilot` replicasets are created. For now we can just leave this as is.

Comment (pbr0ck3r) [reply pkg/kor/replicaset.go:17]: GKE includes Prometheus for metrics out of the box, so this is probably why these exists.

Comment (yonahd) [reply pkg/kor/replicaset.go:17]: Let's remove this for now.
I'm not sure why they would be install empty replicasets and in any case I don't see harm in cleaning it. The deployment will recreate one when it is scaled up

**Review by yonahd — COMMENTED** [review]
A couple of small comments

**Review by yonahd — APPROVED** [review]
Lgtm
Thank you!

**Pull request merged**